### ISSUE TITLE
False positives when destructuring with rename toRef props in vue/no-dupe-keys rule

### ISSUE THREAD

**Issue by N-Silbernagel** [vuejs/eslint-plugin-vue#2197]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: 8.41.0** 
- **eslint-plugin-vue version: 9.14.0** 
- **Node version: 16.14.2** 
- **Operating System: MacOs / StackBlitz** 

**Please show your full configuration:**

```json5
module.exports = {
  extends: [
    '@vue/typescript/recommended',
    'eslint:recommended',
    'plugin:vue/vue3-recommended',
  ],
  rules: {
    // override/add rules settings here, such as:
    // 'vue/no-unused-vars': 'error'
  },
};
```

**What did you do?**	

```vue
<script setup lang="ts">
import { computed, ref, toRefs } from 'vue';

const props = defineProps<{ msg: string }>();

const { msg: msgProp } = toRefs(props);

const msg = computed(() => msgProp.value || 'hello');
</script>
```
This is similar to https://github.com/vuejs/eslint-plugin-vue/pull/2189 but with a renamed prop in destructuring assignment.

**What did you expect to happen?**
No errors.

**What actually happened?**

/src/components/HelloWorld.vue
  8:7  error  Duplicated key 'msg'  vue/no-dupe-keys

✖ 1 problem (1 error, 0 warnings)

**Repository to reproduce this issue**

https://stackblitz.com/edit/vitejs-vite-49r5se?file=src%2Fcomponents%2FHelloWorld.vue

**Comment by FloEdelmann** [comment]
This is actually a valid warning of the `no-dupe-keys` rule: You have both a prop and a computed variable with the same name.
